proposals/Layer-1.md

```diff
@@ -0,0 +1,390 @@
+# Layer 1 exception handling
+
+Layer 1 of exception handling is the MVP (minimal viable proposal) for
+implementing exceptions in WebAssembly. As such, it doesn't include higher-level
+concepts and structures. These concept and structures are introduced in later
+layers, and either:
+
+1. Improves readability by combining concepts in layer 1 into higher-level
+   constructs, thereby reducing code size.
+
+2. Allow performance improvements in the VM.
+
+3. Introduce additional new functionality not available in layer 1.
+   
```

Comment:
We have a list here of what the proposal does not include, but it'd be good to include the requirements for the MVP. To me, these include:
1. Sufficient to implement C++ exception semantics by translating from Clang-generated LLVM code.
2. Allows WebAssembly to throw exceptions that can be caught by JavaScript.
3. Allows WebAssembly to catch exceptions thrown by JavaScript.
4. Can be implemented with low performance overhead.

The performance goal could be made more precise. I think it's roughly "code that does not use exceptions does not run any slower once the exception proposal has been adopted."

Reply:
I opened #34 to track this.